docs/dev.md

```diff
@@ -0,0 +1,15 @@
+#### dev
+
+----
+Start a Liberty server in dev mode.
```

Comment:
Mention that this goal also runs the `create-server`, `install-feature`, and `install-apps` goals before starting the server.